src/components/textarea/textarea.test.tsx

```diff
@@ -0,0 +1,719 @@
+import React, { useState } from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import * as floatingUi from "@floating-ui/react-dom";
+
+import { testStyledSystemMargin } from "../../__spec_helper__/__internal__/test-utils";
+import Textarea, { TextareaProps } from ".";
+import { EnterKeyHintTypes } from "../../__internal__/input";
+import guid from "../../__internal__/utils/helpers/guid";
+import CarbonProvider from "../carbon-provider/carbon-provider.component";
+import Logger from "../../__internal__/utils/logger";
+import StyledInput from "../../__internal__/input/input.style";
+
+jest.mock("../../__internal__/utils/logger");
+
+jest.mock("../../__internal__/utils/helpers/guid");
+const mockedGuid = "guid-12345";
+(guid as jest.MockedFunction<typeof guid>).mockImplementation(() => mockedGuid);
+
+let loggerSpy: jest.SpyInstance<void, [message: string]> | jest.Mock;
+
+beforeEach(() => {
+  loggerSpy = jest.spyOn(Logger, "deprecate");
+});
+
+afterEach(() => {
+  loggerSpy.mockRestore();
+});
+
+afterAll(() => {
+  loggerSpy.mockClear();
+});
+
+test("should display deprecation warning once when rendered as uncontrolled", () => {
+  render(<Textarea name="my-textarea" defaultValue="test" />);
+
+  expect(loggerSpy).toHaveBeenCalledWith(
+    "Uncontrolled behaviour in `Textarea` is deprecated and support will soon be removed. Please make sure all your inputs are controlled."
+  );
+
+  expect(loggerSpy).toHaveBeenCalledTimes(1);
+});
+
+test("should render a textarea element", () => {
+  render(<Textarea />);
+  const textarea = screen.getByRole("textbox");
+  expect(textarea).toBeInTheDocument();
+  expect(textarea.tagName).toEqual("TEXTAREA");
+});
+
+test("should not render character counter if no characterLimit prop is given", () => {
+  render(<Textarea />);
+  expect(screen.queryByText(/you can enter up to/i)).not.toBeInTheDocument();
+});
+
+test("should render character counter if characterLimit prop is given", () => {
+  render(<Textarea characterLimit={5} />);
+  expect(screen.getByText(/you can enter up to/i)).toBeInTheDocument();
+});
+
+test("should not render a placeholder if disabled", () => {
+  render(<Textarea placeholder="foo" disabled />);
+  expect(screen.getByRole("textbox")).toHaveAttribute("placeholder", "");
+});
+
+testStyledSystemMargin((props) => <Textarea {...props} />);
+
+// not sure what this is really testing - I assume it's the expandTextArea stuff, but can't get it to fail!
+// would rather remove this but leave it, commented out, to see if it causes coverage problems. Would like
+// to rewrite if it does, otherwise happy to remove it!
+test("the height of the textarea should remain unchanged when the window is resized", () => {
+  render(<Textarea />);
```

Comment:
comment: yes this is to test that without the `expandable` prop set the textarea height doesn't change when the window resizes and the content wraps, it adds a scrollbar instead etc. When `expandable` is true the height adapts as the content wraps. It might be worth adding text to the input for this test 🤷

Reply:
oops, that was a comment I put in to myself while converting the tests, and forgot to remove it 🤦‍♂️ 

I'll get the comment removed or fixed to reflect actual reality!

Reply:
OK, it turns out this test is fine, and is providing value in that I can get it to fail if I comment out the appropriate `if (expandable)` condition (I think I was trying with the wrong one previously 🤦 ). So I'll leave this in, remove the confusing comment, and move the test so that it's next to the related ones where the `expandable` prop is true, rather than miles before either of them.

While I'm at it I'm also replacing `window.dispatchEvent("resize")` with `fireEvent.resize(window)` - `userEvent` doesn't seem to have anything for resize, and given that I think `fireEvent` is fine to use and is a bit more readable (I think it just does a `dispatchEvent` itself, with some sensible default properties).